### ISSUE TITLE
Clang parsing error in testcomp

### ISSUE THREAD

**Issue by filipeom** [OCamlPro/owi#386]
With clang 16:

```sh
dune exec -- owi c bench/testcomp/sv-benchmarks/c/ldv-regression/test22-3.c 
[parse exception: bad local.get index (at 0:209)]
Fatal: error parsing wasm
clang: error: linker command failed with exit code 1 (use -v to see invocation)
run ['/usr/bin/clang' '-O3' '--target=wasm32' '-m32' '-ffreestanding'
     '--no-standard-libraries' '-Wno-everything' '-flto=thin'
     '-Wl,--entry=main' '-Wl,--export=main' '-Wl,--lto-O0'
     '-Wl,-z,stack-size=8388608' '-I'
     '/home/filipe/lib/owi/_build/install/default/share/owi/libc' '-o'
     'a.out.wasm'
     '/home/filipe/lib/owi/_build/install/default/share/owi/binc/libc.wasm'
     'bench/testcomp/sv-benchmarks/c/ldv-regression/test22-3.c']: exited with 1%  
```

In `results-testcomp-owi_w24_O3_sZ3-2024-07-17_00h31m16` this previously passed

**Comment by filipeom** [comment]
Compiling everything with -O0 seems to fix the parsing error, but:

```sh
dune exec -- owi c bench/testcomp/sv-benchmarks/c/ldv-regression/test22-3.c -O0
All OK%              
```

**Comment by filipeom** [comment]
Oh ok, it's because inlining `_or` in https://github.com/OCamlPro/owi/commit/da85f7bb8355982721846812b40ae6117274da8b uses local that don't exist

**Comment by zapashcanon** [comment]
Fixed by #388.
